zilencer/views.py

```diff
@@ -1190,6 +1191,7 @@ def remote_server_post_analytics(
             value=row.value,
         )
         for row in realm_counts
+        if row.property not in LOGGING_COUNT_STAT_PROPERTIES_NOT_SENT_TO_BOUNCER
```

Comment:
So a problem with this implementation approach if a client happens to have its highest ID row be one of these ignored stats, the client will keep sending the last few values even though it in theory has them.

Probably mostly OK since the effect size should be small, and `realm_active_humans::day` is the last listed `CountStat`, so probably none of these will be the last in typical use, and in any case, it should only be at most a handful of resubmitted rows.

But probably we should add a block comment about this detail in the `LOGGING_COUNT_STAT_PROPERTIES_NOT_SENT_TO_BOUNCER` definition for visibility, maybe with a `TODO` note that a better algorithm would have the `RemoteZulipServer` table store the highest ID that it's processed, rather than querying that directly from what rows exist.

```
def get_last_id_from_server(server: RemoteZulipServer, model: Any) -> int:                              
    last_count = (                                                                                      
        model.objects.filter(server=server)                                                             
        # Rows with remote_id=None are managed by the bouncer service itself,                           
        # and thus aren't meant for syncing and should be ignored here.                                 
        .exclude(remote_id=None)                                                                        
        .order_by("remote_id")                                                                          
        .only("remote_id")                                                                              
        .last()                                                                                         
    )                                                                                                   
    if last_count is not None:                                                                          
        return last_count.remote_id                                                                     
    return 0                                                                                            
```

Reply:
Repushed with a comment and TODO.